# Review

Graph representation learning in biomedicine and healthcare (*Nature Biomedical Engineering* 2022.10) [[Paper](https://www.nature.com/articles/s41551-022-00942-x)\]

Learning functional properties of proteins with language models (*Nature Machine Intelligence* 2022.04) [[Paper](https://www.nature.com/articles/s42256-022-00457-9)\] [[Code](https://github.com/kansil/PROBE)\]





# Sequence-based

## 2022

[**CTR**] Clustered tree regression to learn protein energy change with mutated amino acid (*Briefings in Bioinformatics* 2022.11) [[Paper](https://academic.oup.com/bib/article-abstract/23/6/bbac374/6702668?redirectedFrom=fulltext)\] [[Code](https://github.com/LiamT01/CTR)\]

[**MIF-ST**] Masked inverse folding with sequence transfer for protein representation learning (*BioRxiv* 2022.11) [[Paper](https://www.biorxiv.org/content/10.1101/2022.05.25.493516v2.abstract)\]

[**CARP**] Convolutions are competitive with transformers for protein sequence pretraining (*BioRxiv* 2022.11) \[[Paper](https://www.biorxiv.org/content/10.1101/2022.05.19.492714v3.abstract)] [[Code](https://github.com/microsoft/protein-sequence-models)]

[**PLMs**] Exploring evolution-aware & -free protein language models as protein function predictors (*NeurIPS* 2022.10) \[[Paper](https://openreview.net/forum?id=U8k0QaBgXS)] [[Code](https://github.com/elttaes/Revisiting-PLMs)]

[**ESM-2**] Language models of protein sequences at the scale of evolution enable accurate structure prediction (*BioRxiv* 2022.10) [[Paper](https://www.biorxiv.org/content/10.1101/2022.07.20.500902v2.abstract)\] \[[Code](https://github.com/facebookresearch/esm)\]

[**I-TASSER-MTD**] I-TASSER-MTD: a deep-learning-based platform for multi-domain protein structure and function prediction (*Nature Protocols* 2022.08) [[paper\]](https://www.nature.com/articles/s41596-022-00728-0) \[[Code](https://zhanggroup.org/I-TASSER-MTD/)\]

[**ATOMRefine**] Atomic protein structure refinement using all-atom graph representations and SE(3)-equivariant graph neural networks (*BioRxiv* 2022.08) [[paper\]](https://www.biorxiv.org/content/10.1101/2022.05.06.490934v2.abstract)

[**Cytoself**] Self-supervised deep learning encodes high-resolution features of protein subcellular localization (*Nature Methods* 2022.07) [[Paper\]](https://www.nature.com/articles/s41592-022-01541-z) [[Code](https://github.com/royerlab/cytoself)]

[**ProtGPT2**] ProtGPT2 is a deep unsupervised language model for protein design (*Nature Communications* 2022.07) [[Paper\]](https://www.nature.com/articles/s41467-022-32007-7)[[Code](https://huggingface.co/docs/transformers/main_classes/trainer)]

[**OmegaFold**] High-resolution de novo structure prediction from primary sequence (*BioRxiv* 2022.07) \[[Paper](https://www.biorxiv.org/content/10.1101/2022.07.21.500999v1.abstract)]

[**generative_IDPs**] Artificial Intelligence Guided Conformational Mining of Intrinsically Disordered Proteins (*Communications Biology* 2022.06) \[[Paper](https://www.nature.com/articles/s42003-022-03562-y)] [[Code](https://github.com/aaayushg/generative_IDPs)]

[**Effector-GAN**] Effector-GAN: prediction of fungal effector proteins based on pretrained deep representation learning methods and generative adversarial networks (*Bioinformatics* 2022.05) \[[Paper](https://academic.oup.com/bioinformatics/article-abstract/38/14/3541/6596046?redirectedFrom=fulltext&login=false)] [[Code](http://47.109.24.44:4747/wangys_biolab/effector-gan/)] [[Web server](http://lab.malab.cn/~wys/webserver/Effector-GAN)]

Learning meaningful representations of protein sequences (*Nature Communications* 2022.04) [[Paper](nature.com/articles/s41467-022-29443-w)\] [[Code](https://github.com/MachineLearningLifeScience/meaningful-protein-representations)\]

[**Evolocity**] Evolutionary velocity with protein language models predicts evolutionary dynamics of diverse proteins (*Cell Systems* 2022.04) \[[Paper](https://www.sciencedirect.com/science/article/pii/S2405471222000382)] [[Code](https://github.com/brianhie/evolocity)] \[[Chinese blog](http://www.360doc.com/content/22/0621/17/71609523_1036895157.shtml)\]

[**ProteinBERT**] ProteinBERT: a universal deep-learning model of protein sequence and function (*Bioinformatics* 2022.04) \[[Paper](https://academic.oup.com/bioinformatics/article/38/8/2102/6502274)] [[Code](https://github.com/nadavbra/protein_bert)]

Using deep learning to annotate the protein universe (**Nature Biotechnolog** 2022.02) [[Paper](https://www.nature.com/articles/s41587-021-01179-w)\] [[Code](https://github.com/google-research/google-research/tree/master/using_dl_to_annotate_protein_universe)\]



## 2021

[**trRosetta**] The trRosetta server for fast and accurate protein structure prediction (*Nature protocols* 2021.11) \[[Web server](https://yanglab.nankai.edu.cn/trRosetta/)]

[**PMLM**] Pre-training co-evolutionary protein representation via a pairwise masked language model (*Arxiv* 2021.10) \[[Paper](https://arxiv.org/abs/2110.15527)]

[**ESM-1v**] Language models enable zero-shot prediction of the effects of mutations on protein function (*NeurIPS* 2021.09) \[[Paper](https://proceedings.neurips.cc/paper/2021/hash/f51338d736f95dd42427296047067694-Abstract.html)] [[Code](https://github.com/facebookresearch/esm)]

[**NeuroSEED**] Neural Distance Embeddings for Biological Sequences (*NeurIPS* 2021.09) [[Paper\]](https://proceedings.neurips.cc/paper/2021/hash/9a1de01f893e0d2551ecbb7ce4dc963e-Abstract.html) [[Code](https://github.com/gcorso/NeuroSEED)]

[**MSA Transformer**] MSA Transformer (*ICML* 2021.09) [[Paper](http://proceedings.mlr.press/v139/rao21a.html)] [[Code](https://github.com/facebookresearch/esm)]

[**RoseTTAFold**] Accurate prediction of protein structures and interactions using a three-track neural network (*Science* 2021.08) [[Paper\]](https://www.science.org/doi/abs/10.1126/science.abj8754) [[Code](https://github.com/RosettaCommons/RoseTTAFold)] [[pdf](https://www.ipd.uw.edu/wp-content/uploads/2021/07/Baek_etal_Science2021_RoseTTAFold.pdf)]

[**ProteinLM**] Modeling Protein Using Large-scale Pretrain Language Model (*KDD: the International Workshop on  Pretraining: Algorithms, Architectures, and Applications* 2021.08) [[Paper](https://arxiv.org/abs/2108.07435)] [[Code](https://github.com/THUDM/ProteinLM)]

[**AlphaAFold**] Highly accurate protein structure prediction with AlphaFold (*Nature* 2021.07) [[Paper\]](https://www.nature.com/articles/s41586-021-03819-2) [[Code](https://github.com/deepmind/alphafold)]

[**pLMs**] ProtTrans: Towards Cracking the Language of Lifes Code Through Self-Supervised Learning (*TPAMI* 2021.07) [[Paper](https://ieeexplore.ieee.org/abstract/document/9477085)] [[Code](https://github.com/agemagician/ProtTrans)]

Learning the protein language: Evolution, structure, and function (*Cell Systems* 2021.06) [[Paper](https://www.sciencedirect.com/science/article/pii/S2405471221002039)] [[Code](https://github.com/tbepler/prose)]

Combining evolutionary and assay-labelled data for protein fitness prediction (*bioRxiv* 2021.03) [[Paper](biorxiv.org/content/10.1101/2021.03.28.437402v1.abstract)]



## 2019

[**TAPE**] Evaluating Protein Transfer Learning with TAPE (*NeurIPS* 2019.09) \[[Paper](https://proceedings.neurips.cc/paper/2019/hash/37f65c068b7723cd7809ee2d31d7861c-Abstract.html)] [[Code](https://github.com/songlab-cal/tape)] [[Chinese blog](https://mp.weixin.qq.com/s/WCCZiajDplcrkKGiy9PC2g)]





# Graph-based

## 2022

[**DeepRank-GNN**] DeepRank-GNN: A Graph Neural Network Framework to Learn Patterns in Protein-Protein Interfaces (*Bioinformatics* 2022.11) \[[Paper](https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btac759/6845451)] [[Code](https://github.com/DeepRank/DeepRank-GNN)]

[**GearNet**] Protein representation learning by geometric structure pretraining (*Arxiv* 2022.09) \[[Paper](https://arxiv.org/abs/2203.06125)] [[Code](https://github.com/DeepGraphLearning/GearNet)]

[**DW-GNN**] Directed Weight Neural Networks for Protein Structure Representation Learning (*Arxiv* 2022.09) \[[Paper](https://arxiv.org/abs/2201.13299)]

[**GBPNet**] GBPNet: Universal Geometric Representation Learning on Protein Structures (*KDD* 2022.08) \[[Paper](https://dl.acm.org/doi/abs/10.1145/3534678.3539441)] [[Code](https://github.com/sarpaykent/GBPNet)] [[pdf](https://scholar.archive.org/work/wyyhzybbsrftlgibtv325wm3ni/access/wayback/https://dl.acm.org/doi/pdf/10.1145/3534678.3539441)]

[**STEPS**] Structure-aware Protein Self-supervised Learning (*Arxiv* 2022.06) \[[Paper](https://arxiv.org/abs/2204.04213)]

[**GNN_F, GNN_P**] Decoding the protein–ligand interactions using parallel graph neural networks (*Scientific Reports* 2022.05) [[Paper](nature.com/articles/s41598-022-10418-2)\] [[Code](https://github.com/PNNL-CompBio/pf-gnn_pli)\]

[**CRL3DProt**] Contrastive Representation Learning for 3D Protein Structures (*Arxiv* 2022.05) \[[Paper](https://arxiv.org/abs/2205.15675)]

[**BACPI**] BACPI: a bi-directional attention neural network for compound–protein interaction and binding affinity prediction (*Bioinformatics* 2022.04) [[Paper](https://academic.oup.com/bioinformatics/article-abstract/38/7/1995/6511437?redirectedFrom=fulltext)\] [[Code](https://github.com/CSUBioGroup/BACPI)\]

[**STAMP-DPI**] Structure-Aware Multimodal Deep Learning for Drug–Protein Interaction Prediction (*JCIM* 2022.04) \[[Paper](https://pubs.acs.org/doi/abs/10.1021/acs.jcim.2c00060)] \[[Code](https://github.com/biomed-AI/STAMP-DPI)]

[**GraphSite**] Fast AlphaFold2-aware protein–DNA binding site prediction using graph transformer (*Briefings in Bioinformatics* 2022.03) \[[Paper](https://academic.oup.com/bib/article-abstract/23/2/bbab564/6509729)] \[[Code](https://github.com/biomed-AI/GraphSite)] \[[Web server](https://biomed.nscc-gz.cn/apps/GraphSite)]

[**GraSR**] Fast protein structure comparison through effective representation learning with contrastive graph neural networks (*PLOS Computational Biology* 2022.03) \[[Paper](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1009986)] \[[Code](https://github.com/chunqiux/GraSR)]



## 2021

[**InteractionGraphNet**] InteractionGraphNet: A Novel and Efficient Deep Graph Representation Learning Framework for Accurate Protein−Ligand Interaction Predictions (*JMC* 2021.12) [[Paper](https://pubs.acs.org/doi/pdf/10.1021/acs.jmedchem.1c01830)] [[Code](https://github.com/zjujdj/InteractionGraphNet/tree/master)]

[**nn4dms**] Neural networks to learn protein sequence–function relationships from deep mutational scanning data (*PNAS* 2021.11) [[Paper](https://www.pnas.org/doi/abs/10.1073/pnas.2104878118)] [[Code](https://github.com/gitter-lab/nn4dms)]

[**PG-GNN**] Geometric Graph Representation Learning on Protein Structure Prediction (*KDD* 2021.08) [[Paper](https://dl.acm.org/doi/abs/10.1145/3447548.3467323)]

Distillation of MSA Embeddings to Folded Protein Structures with Graph Transformers (*bioRxiv* 2021.06) \[[Paper](https://www.biorxiv.org/content/10.1101/2021.06.02.446809.abstract)]

[**DeepFRI**] Structure-based protein function prediction using graph convolutional networks (*Nature Communications* 2021.05) \[[Paper](https://www.nature.com/articles/s41467-021-23303-9)] \[[Code](https://github.com/flatironinstitute/DeepFRI)]

[**PtsRep**] Self-Supervised Representation Learning of Protein Tertiary Structures (PtsRep) and Its Implications for Protein Engineering (*bioRxiv* 2021.03) [[Paper](https://www.biorxiv.org/content/10.1101/2020.12.22.423916v2.abstract)]

[**GraphQA**] GraphQA: protein model quality assessment using graph convolutional networks (*Bioinformatics* 2021.02) \[[Paper](https://academic.oup.com/bioinformatics/article/37/3/360/5891171)] \[[Code](https://github.com/baldassarreFe/protein-quality-gn)]

[**GVP**] Learning from Protein Structure with Geometric Vector Perceptrons (*ICLR* 2021.01) \[[Paper](https://openreview.net/forum?id=1YLJDvSx6J4)] \[[Code](https://github.com/drorlab/gvp-pytorch)]

[**IEConv_proteins**] Intrinsic-Extrinsic Convolution and Pooling for Learning on 3D Protein Structures (*ICLR* 2021.01) \[[Paper](https://openreview.net/forum?id=l0mSUROpwY)] \[[Code](https://github.com/phermosilla/IEConv_proteins)]



# Vision-based

## 2022

[**masif-seed**] *De novo* design of site-specific protein interactions with learned surface fingerprints (*bioArxiv* 2022.06) \[[Paper](https://www.biorxiv.org/content/10.1101/2022.06.16.496402v1.abstract)] \[[Code](https://github.com/LPDI-EPFL/masif_seed)]

[**PointSite**] PointSite: A Point Cloud Segmentation Tool for Identification of Protein Ligand Binding Atoms (*JCIM* 2022.05) \[[Paper](https://pubs.acs.org/doi/abs/10.1021/acs.jcim.1c01512)] \[[Code](https://github.com/PointSite/PointSite)]

A point cloud-based deep learning strategy for protein–ligand binding affinity prediction (*Briefings in Bioinformatics* 2022.01) \[[Paper](https://academic.oup.com/bib/article-abstract/23/1/bbab474/6440132?redirectedFrom=fulltext&login=false)]



## 2021

[**DeepRank**] DeepRank: a deep learning framework for data mining 3D protein-protein interfaces (*Nature Communications* 2021.12) \[[Paper](https://www.nature.com/articles/s41467-021-27396-0)] \[[Code](https://github.com/DeepRank/deeprank)]

[**ProtConv**] Convolutional neural networks with image representation of amino acid sequences for protein function prediction (*Computational Biology and Chemistry* 2021.06) [[Paper](https://www.sciencedirect.com/science/article/abs/pii/S147692712100061X)] [[Code](https://github.com/swakkhar/ProtConv)]

[**dMaSIF**] Fast End-to-End Learning on Protein Surfaces (*CVPR* 2021.03) \[[Paper](https://openaccess.thecvf.com/content/CVPR2021/html/Sverrisson_Fast_End-to-End_Learning_on_Protein_Surfaces_CVPR_2021_paper.html)] [[Code](https://github.com/FreyrS/dMaSIF/tree/master)]



# Multi modality-based

## 2022

[**CPAC**] Cross-Modality and Self-Supervised Protein Embedding for Compound–Protein Affinity and Contact Prediction (*Bioinformatics* 2022.09) [[Paper\]](https://academic.oup.com/bioinformatics/article/38/Supplement_2/ii68/6702011)[[Code](https://github.com/Shen-Lab/CPAC)]

[**OntoProtein**] OntoProtein: Protein Pretraining With Gene Ontology Embedding (*ICLR* 2022.01) [[Paper\]](https://arxiv.org/abs/2201.11147)[[Code](https://github.com/zjunlp/OntoProtein)]



## 2021

[**LM-GVP**] NeuralLM-GVP: A Generalizable Deep Learning Framework for Protein Property Prediction from Sequence and Structure (*bioRxiv* 2021.09) [[Paper](https://www.biorxiv.org/content/10.1101/2021.09.21.460852v1.abstract)] [[Code](https://github.com/aws-samples/lm-gvp)]

[**HOLOPROT**] Multi-Scale Representation Learning on Proteins (*NeurIPS* 2021.09) \[[Paper](https://proceedings.neurips.cc/paper/2021/hash/d494020ff8ec181ef98ed97ac3f25453-Abstract.html)] \[[Code](https://github.com/vsomnath/holoprot)]

Toward More General Embeddings for Protein Design: Harnessing Joint Representations of Sequence and Structure (*bioRxiv* 2021.09) \[[Paper](https://www.biorxiv.org/content/10.1101/2021.09.01.458592v1.abstract)]

Unsupervised protein embeddings outperform hand-crafted sequence and structure features at predicting molecular function (*Bioinformatics* 2021.01) \[[Paper](https://academic.oup.com/bioinformatics/article/37/2/162/5892762)] \[[Code](https://github.com/stamakro/GCN-for-Structure-and-Function)]

